microsoft-edge/progressive-web-apps-chromium/how-to/index.md

```diff
@@ -0,0 +1,319 @@
+
+In this overview of PWA basics, you create a simple web app and extend it as a PWA.  The finished project works across modern browsers.
+
+> [!TIP]
+> You can use [PWABuilder][PwaBuilder] to create a new PWA, enhance your existing PWA, or package your PWA for app stores.  
+
+
+<!-- ====================================================================== -->
+## Prerequisites
+
+*   Use [Visual Studio Code][VisualstudioCodeMain] to edit your PWA source code.
+*   Use [Node.js][NodejsMain] as your local web server.
+
+
+<!-- ====================================================================== -->
+## Creating a basic web app
+
+To create an empty web app, follow the steps in [Node Express App Generator][ExpressjsApplicationGenerator], and name your app `MySamplePwa`.
+
+In the prompt, run the following commands.
+
+```shell
+npx express-generator --no-view
+```
+
+```shell
+npm install
+```
+
+The commands create an empty web app and install any dependencies.
+
+You now have a simple, functional web app.  To start your web app, run the following command.
+
+```shell
+npm start
+```
+
+Now browse to `http://localhost:3000` to view your new web app.
+
+:::image type="complex" source="../media/visual-studio-nodejs-express-index.png" alt-text="Running your new PWA on localhost" lightbox="../media/visual-studio-nodejs-express-index.png":::
+   Running your new PWA on localhost
+:::image-end:::
+
+
+<!-- ====================================================================== -->
+## Getting started building a PWA
+
+Now that you have a simple web app, extend it as a PWA by adding the three requirements for PWAs: [HTTPS](#step-1---use-https), a [Web App Manifest](#step-2---create-a-web-app-manifest), and a [Service Worker](#step-3---add-a-service-worker).
+
+
+<!-- ====================================================================== -->
+## Step 1 - Use HTTPS
+
+Key parts of the PWA platform, such as [Service Workers][MDNServiceWorkerApi], require the use of HTTPS.  When your PWA goes live, you must publish it to an HTTPS URL.  Many hosts now offer HTTPS by default, but if your host doesn't, Let's Encrypt offers a free alternative for creating the necessary certificates.
+
+For debugging purposes, Microsoft Edge also permits `http://localhost` to use the PWA APIs.  In this tutorial, you use `http://localhost` to build your PWA.
+
+[Publish your web app as a live site][VisualStudioNodejsTutorialPublishAzureAppService], but make sure your server is configured for HTTPS.  For example, you can create an [Azure free account][AzureCreateFreeAccount].  Host your site on the [Microsoft Azure App Service][AzureWebApps] and it is served over HTTPS by default.  
```

Comment:
I understand why we might not want to reference Netlify or other competitors, but do we want to offer up GitHub hosting as an option here too?

Reply:
Yeah, that seems like a good option to me. In fact because this page doesn't rely on the site being published at all, and we expect folks to use `http://localhost` I think we could rephrase this as a list of options. Something like: "Microsoft Edge lets you use the PWA features on http://localhost, but once your app is live, you will need an HTTPS URL, here are a few options: GitHub pages, Azure App Service, Let's Encrypt".